internal/controller/serving/servingruntime_controller.go

```diff
@@ -259,24 +265,178 @@ func (r *ServingRuntimeReconciler) Reconcile(ctx context.Context, req ctrl.Reque
+			if apierrs.IsNotFound(err) {
+				if err := r.Client.Create(ctx, rayDefaultSecret); err != nil {
+					logger.Error(err, "fail to create ray tls secret", "secret", constants.RayTLSSecretName, "namespace", targetNamespace)
+					return err
+				}
+			} else {
+				logger.Error(err, "fail to get ray tls secret", "secret", constants.RayTLSSecretName, "namespace", targetNamespace)
+				return err
+			}
+		} else {
+			// If the ca secret is updated, it updates the ca cert in default ray tls secret in the user namespace
+			if !bytes.Equal(caCertSecret.Data[corev1.TLSCertKey], defaultCertSecret.Data[rayCaCertNameInUserNS]) {
+				updatedDefaultCertSecret := defaultCertSecret.DeepCopy()
+				updatedDefaultCertSecret.Data[rayCaCertNameInUserNS] = caCertSecret.Data[corev1.TLSCertKey]
+				if err := r.Client.Update(ctx, updatedDefaultCertSecret); err != nil {
+					logger.Error(err, "fail to update ray tls secret", "secret", constants.RayTLSSecretName, "namespace", targetNamespace)
+					return err
+				}
+			}
+		}
+	} else {
+		if !utils.IsRayTLSSecret(reqName) {
+			if err := r.Client.Delete(ctx, rayDefaultSecret); err != nil {
+				if apierrs.IsNotFound(err) {
+					return nil
+				}
+				logger.Error(err, "fail to delete ray tls secret", "secret", constants.RayTLSSecretName, "namespace", targetNamespace)
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+func getDesiredRayDefaultSecret(namespace string, caSecret *corev1.Secret) *corev1.Secret {
+	return &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      constants.RayTLSSecretName,
+			Namespace: namespace,
+			Labels: map[string]string{
+				"opendatahub.io/managed":       "true",
+				"app.kubernetes.io/name":       "self-signed-ray-cert",
+				"app.kubernetes.io/component":  "odh-model-serving",
+				"app.kubernetes.io/part-of":    "kserve",
+				"app.kubernetes.io/managed-by": "odh-model-controller",
+			},
+		},
+		Data: map[string][]byte{
+			rayCaCertNameInUserNS: caSecret.Data[corev1.TLSCertKey],
+		},
+		Type: corev1.SecretTypeOpaque,
+	}
+}
+
+func existMultiNodeServingRuntimeInNs(srList servingv1alpha1.ServingRuntimeList) bool {
+	existMultiNodeServingRuntime := false
+	for _, sr := range srList.Items {
+		existMultiNodeServingRuntime = isMultiNodeServingRuntime(sr.Name)
+	}
+	return existMultiNodeServingRuntime
+}
+
+// Determine if ServingRuntime matches specific conditions
+// TO-DO upstream Kserve 0.15 will have a new API WorkerSpec
+// So for now, it will check servingRuntime name, but after we move to 0.15, it needs to check workerSpec is specified or not.(RHOAIENG-16147)
+func isMultiNodeServingRuntime(srName string) bool {
+	return srName == "vllm-multinode-runtime"
+}
+
 // SetupWithManager sets up the controller with the Manager.
 func (r *ServingRuntimeReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&servingv1alpha1.ServingRuntime{}).
 		Named("servingruntime").
+		Owns(&k8srbacv1.RoleBinding{}).
```

Comment:
No longer needed?

Reply:
this is not for my pr, it is for existing metrics logic.